pvsystemprofiler/estimator.py

```diff
@@ -52,34 +58,35 @@ def __init__(self, data_handler, gmt_offset):
         self.eot_da_rosa = eot_da_rosa(self.day_of_year)
         self.delta = None
         self.omega = None
+        if day_selection_method == 'all':
+            self.days = self.data_handler.daily_flags.no_errors
+        elif day_selection_method == 'clear':
+            self.days = self.days.daily_flags.clear
+        elif day_selection_method == 'cloudy':
+            self.days = self.data_handler.daily_flags.cloudy
+
+        if solar_noon_method == 'optimized_estimates' or daylight_method == 'optimized_estimates':
+            ss = SunriseSunset()
 
+        if solar_noon_method == 'rise_set_average':
+            self.solarnoon = avg_sunrise_sunset(self.data_matrix)
+        elif solar_noon_method == 'energy_com':
+            self.solarnoon = energy_com(self.data_matrix)
+        if solar_noon_method == 'optimized_estimates':
+            ss.run_optimizer(data=self.data_matrix)
+            self.solarnoon = np.nanmean([ss.sunrise_estimates, ss.sunset_estimates], axis=0)
+
+        if daylight_method in ('sunrise-sunset', 'sunrise sunset'):
+            self.hours_daylight = calculate_hours_daylight(self.data_matrix, daytime_threshold)
+        elif daylight_method == 'optimized_estimates':
+            self.hours_daylight = ss.sunset_estimates - ss.sunrise_estimates
+
+    # estimate longitude
     def estimate_longitude(self, estimator='fit_l1',
```

Comment:
data filtering should happen in this function, not in lower-level helper functions